blob/s3blob/s3blob.go

```diff
@@ -71,30 +56,53 @@ import (
 const defaultPageSize = 1000
 
 func init() {
-	blob.Register("s3", openURL)
+	blob.DefaultURLMux().RegisterBucket(Scheme, new(lazySessionOpener))
 }
 
-func openURL(ctx context.Context, u *url.URL) (driver.Bucket, error) {
-	q := u.Query()
-	cfg := &aws.Config{}
+// URLOpener opens S3 URLs like "s3://mybucket".
+type URLOpener struct {
+	// ConfigProvider must be set to a non-nil value.
+	ConfigProvider client.ConfigProvider
 
-	if region := q["region"]; len(region) > 0 {
```

Comment:
Can you put these back into the default implementation?

Reply:
I failed to follow up on this comment. I think it's important to include these, will send a PR.